zapcore/sampler.go

```diff
@@ -38,6 +47,37 @@ type counter struct {
 
 type counters [_numLevels][_countersPerLevel]counter
 
+// Report defines a function that generates and writes sampling activity report
+// log entries.
+type Report func(core Core, ent Entry, n int64) error
```

Comment:
In my view, any exported `type T func()` being added, especially to a post-1.0 stable project, should immediately be re-written as an interface:
```
type Ter interface {
  thing()
}
```
with maybe a convenience
```
type tFunc func()
func (tf tFunc) thing() { tf() }
```